app/src/androidTest/java/com/sdp13epfl2021/projmag/database/FakeDatabaseTest.kt

```diff
@@ -0,0 +1,102 @@
+package com.sdp13epfl2021.projmag.database
+
+import com.sdp13epfl2021.projmag.model.ImmutableProject
+import java.util.*
+
+
+class FakeDatabaseTest(projectsBeginning: List<ImmutableProject>) : ProjectsDatabase {
+
+    var projects: List<ImmutableProject> = projectsBeginning
+    var listeners: List<((ProjectChange) -> Unit)> = emptyList()
+    var nextId: Int = 90000

```

Comment:
ugly solution

Reply:
This class is only used in tests, as a fake db to provide a ProjectDatabase.
The nextId in firebase is random and unique, here I only used a counter because only a few known projects are added.